### PR TITLE
Document new environment variables and display options

### PR DESCRIPTION
**Pull request by sawmurai**
Pull request for ticket https://github.com/symfony/symfony-docs/issues/7177

### CODE REVIEW

File: components/var_dumper/advanced.rst [hunk 1]
```diff
@@ -97,7 +97,12 @@ Dumpers
 
 A dumper is responsible for outputting a string representation of a PHP variable,
 using a :class:`Symfony\\Component\\VarDumper\\Cloner\\Data` object as input.
-The destination and the formatting of this output vary with dumpers.
+The destination and the formatting of this output vary with dumpers and are 
+influenced by two environment variables: 
+If ``DUMP_STRING_LENGTH`` is set, then
+the length of a string is displayed next to its content. 
+If ``DUMP_LIGHT_ARRAY`` is set, 
+then arrays are not displayed completely.
```
Comment (xabbuh) [inline components/var_dumper/advanced.rst]: Would it make sense to render the two sentences as a list? This would then look like this:

```rst
The destination and the formatting of this output vary with dumpers and are
influenced by two environment variables:

* If ``DUMP_STRING_LENGTH`` is set, then the length of a string is displayed
  next to its content.
* If ``DUMP_LIGHT_ARRAY`` is set, then arrays are not displayed completely.
```

File: components/var_dumper/advanced.rst [hunk 2]
```diff
@@ -181,6 +186,21 @@ method. They also typically implement the
 them from re-implementing the logic required to walk through a
 :class:`Symfony\\Component\\VarDumper\\Cloner\\Data` object's internal structure.
 
+The HTMLDumper limits string length and nesting depth of the output.
```
Comment (xabbuh) [inline components/var_dumper/advanced.rst]: We should enclose HTMLDumper with double backticks and change it to `HtmlDumper` to reflect the actual class name.

File: components/var_dumper/advanced.rst [hunk 3]
```diff
@@ -97,7 +97,7 @@ Dumpers
 
 A dumper is responsible for outputting a string representation of a PHP variable,
 using a :class:`Symfony\\Component\\VarDumper\\Cloner\\Data` object as input.
-The destination and the formatting of this output vary with dumpers.
+The destination and the formatting of this output vary with dumpers.
```
Comment (xabbuh) [inline components/var_dumper/advanced.rst]: Oh, can we please revert this change? :)

File: components/var_dumper/advanced.rst [hunk 4]
```diff
@@ -181,6 +181,94 @@ method. They also typically implement the
...
+    ));
+
+    // Limit nesting to 1 level and string length to 160 characters (default)
+
+The output format of a dumper can be fine tuned by the two flags 
+``DUMP_STRING_LENGTH`` and ``DUMP_LIGHT_ARRAY`` which are passed as a bitmap
+in the third constructor argument. They can also be set via environment 
+variables when using
+:method:`assertDumpEquals($dump, $data, $message) <Symfony\\Component\\VarDumper\\Test\\VarDumperTestTrait::assertDumpEquals>`
+during unit testing. The flags can be configured in ``phpunit.xml.dist``.
```
Comment (xabbuh) [inline components/var_dumper/advanced.rst]: Maybe we should better say "in the PHPUnit configuration" (if you are working on a fork you will probably modify the `phpunit.xml` file to change the behaviour only locally).

File: components/var_dumper/advanced.rst [hunk 5]
```diff
@@ -181,6 +181,94 @@ method. They also typically implement the
...
+
+* If ``DUMP_STRING_LENGTH`` is set, then the length of a string is displayed
+  next to its content.
+
+::
```
Comment (xabbuh) [inline components/var_dumper/advanced.rst]: If you do not want to terminate the last sentence preceding a code example with a colon, you have to be more explicit and have to use `.. code-block:: php` instead of the double colon.

File: components/var_dumper/advanced.rst [hunk 6]
```diff
@@ -181,6 +181,94 @@ method. They also typically implement the
...
+    echo $dumper->dump($var, true);
+
+    // (added string length before the string)
+    // array:1 [
+    //   0 => (4) "test"
+    // ]
+
+* If ``DUMP_LIGHT_ARRAY`` is set, then arrays are dumped in a shortened format.
+
+::
```
Comment (xabbuh) [inline components/var_dumper/advanced.rst]: same here

File: components/var_dumper/advanced.rst (hunk 5 above)
Comment (xabbuh) [inline components/var_dumper/advanced.rst]: Oh, and you will need to indent the whole code example by two additional spaces so that it is part of the list item (same for the other item and code example below).

File: components/var_dumper/advanced.rst [hunk 7]
```diff
@@ -181,6 +181,94 @@ method. They also typically implement the
...
+
+    $dumper = new CliDumper(null, null, AbstractDumper::DUMP_STRING_LENGTH);
+    echo $dumper->dump($var, true);
+
+    // (added string length before the string)
+    // array:1 [
+    //   0 => (4) "test"
+    // ]
+
+* If ``DUMP_LIGHT_ARRAY`` is set, then arrays are dumped in a shortened format.
```
Comment (xabbuh) [inline components/var_dumper/advanced.rst]: Maybe add "similar to PHP's short array notation" in parentheses after "shortened format"?

**Comment by sawmurai** [comment]
Thanks for all the help so far :)

**Review by xabbuh — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]
👍  @sawmurai I like this a lot. Thanks!

I've made some minor formatting changes because list of elements are not very readable when they are long, so I replaced it by regular paragraphs. Thanks!

**Comment by xabbuh** [comment]
Thank you @sawmurai.
